src/m_misc2.h

```diff
@@ -46,4 +48,22 @@ void M_CopyLumpName(char *dest, const char *src);
 char *AddDefaultExtension(char *path, const char *ext);
 void NormalizeSlashes(char *str);
 
+inline static long M_DoubleToIntCeil(double arg)
+{
+    int direction = fegetround();
+    fesetround(FE_UPWARD);
+    long result = lrint(arg);
+    fesetround(direction);
+    return result;
+}
+
+inline static long M_DoubleToIntFloor(double arg)
+{
+    int direction = fegetround();
+    fesetround(FE_DOWNWARD);
+    long result = lrint(arg);
+    fesetround(direction);
+    return result;
+}
```

Comment:
@fabiangreffrath This is technically correct, but there may be a more elegant way. Also, how about moving the config code to `m_config.c` and merging `m_misc.c` and `m_misc2.c`?

Reply:
Sure. Now, or for a later PR?

Reply:
What's wrong with using `(int)ceil()`/`(int)floor()`, though?

Reply:
> Now, or for a later PR

I think later.

> What's wrong with using `(int)ceil()`/`(int)floor()`, though?

GCC doesn't like it. I guess `int` is not enough for `double`.

Reply:
How about `(long)ceil()` then? The approach here appears a bit over-engineered for what it actually does.

Reply:
> How about `(long)ceil()` then? The approach here appears a bit over-engineered for what it actually does.

What I meant here is, this is close to ASM in terms of second-guessing the compiler. Let's just tell him what we want and leave it to him to decide about the actual implementation.

Reply:
Apparently, to get rid of the warning, we shouldn't cast it explicitly, so just `ceil()` is correct.

Reply:
I guess the issue was not about `long` or `int`, but about the fact that you were going to directly cast the result of a function call, right?

Reply:
> I guess the issue was not about `long` or `int`, but about the fact that you were going to directly cast the result of a function call, right?

Yes, I think so. It's just a GCC warning, Clang and MSVC compiled it normally. Personally I don't see what was wrong with this cast, probably a false positive.

Reply:
Maybe we should just remove that warning again. It's the one I added just lately. If it prevents clear code by not allowing to cast function return values, then it's not worth it.

Reply:
> If it prevents clear code by not allowing to cast function return values, then it's not worth it.

I agree.

Reply:
Done on the master branch.

Reply:
Thanks!